rewrites/src/main/scala/impl/Power.scala

```diff
@@ -11,28 +10,18 @@ import scalafix.internal.rule.CompilerException
 // From scalafix-rules/src/main/scala/scalafix/internal/rule/CompilerTypePrinter.scala#L55
 // in scalacenter/scalafix (at db93793c2a8f80d0e7bbcacef729dc5c2fa7ceb8).
 final class Power(g: ScalafixGlobal)(implicit doc: SemanticDocument) {
-  private lazy val unit = g.newCompilationUnit(doc.input.text, doc.input.syntax)
-
   def isJavaDefined(t: Tree): Boolean = try {
-    val meth = gsymbol(t)
-    isJavaDefined1(meth) || meth.overrides.exists(isJavaDefined1)
+    gsymbol(t).overrideChain.exists(sym => sym.isJavaDefined || sym.owner == g.definitions.AnyClass)
   } catch {
     case e: Throwable => throw CompilerException(e)
   }
 
-  private def isJavaDefined1(sym: g.Symbol) =
-    sym.isJavaDefined || sym.owner == g.definitions.AnyClass
-
   private def gsymbol(t: Tree): g.Symbol = {
-    GlobalProxy.typedTreeAt(g, unit.position(t.pos.start))
-
-    val sym = g
-      .inverseSemanticdbSymbols(t.symbol.value)
-      .find(t.symbol.value == g.semanticdbSymbol(_))
-      .getOrElse(g.NoSymbol)
-
-    if (sym.info.exists(g.definitions.NothingTpe == _))
-      sym.overrides.lastOption.getOrElse(sym)
-    else sym
+    val sSym = t.symbol
+    val gSymbols = g.inverseSemanticdbSymbols(sSym.value)
+    val gSym = gSymbols.find(gSym => g.semanticdbSymbol(gSym) == sSym.value).getOrElse(g.NoSymbol)
+    if (gSym.info.exists(g.definitions.NothingTpe == _))
+      gSym.overrides.lastOption.getOrElse(gSym)
```

Comment:
Do you know this special case covers?

Reply:
It comes from https://github.com/scalacenter/scalafix/commit/49cf3ca6880d6d00148378762bb745fd9ce94521, but I'm failing to grok: it talks about "when the `override` method has `Nothing` in the inferred type", hasn't inference finished by now?  Is it `Nothing` because the code is carefully avoid the type to be forced, or because the reset of the type inference happens later in a later step of the type-checker?

Either way, let me know if it would be good to try to drop this or change it.

Reply:
It traverses all components of the symbol's type, and if anything refers to `Nothing`, it takes the root overridden method. I have no idea why, maybe to get rid of some `???` overrides?

I'm fine either way, was just wondering if you knew more.

```
scala> class A { override def toString = ??? }

scala> val gSym = typeOf[A].member(TermName("toString"))
val gSym: $r.intp.global.Symbol = method toString

scala> gSym.owner
val res9: $r.intp.global.Symbol = class A

scala> val r = if (gSym.info.exists(definitions.NothingTpe == _)) gSym.overrides.lastOption.getOrElse(gSym) else gSym
val r: $r.intp.global.Symbol = method toString

scala> r.owner
val res10: $r.intp.global.Symbol = class Any
```

Reply:
IIRC, the motivation for this workaround was to avoid inserting `Nothing` into the source code. In the case of `ExplicitResultTypes`, there is no good answer on whether it's best to insert the type signature of the submethod or the supermethod. Both approaches have their pros and cons.

I don't think this workaround is needed in this PR here.

Reply:
Ah, I see.